### PR TITLE
luminous-ttv: init at 0.5.7

### PR DESCRIPTION
**Pull request by alexanderjkslfj**
## Description of changes

Add new package: luminous-ttv
A Rust server to retrieve and relay a playlist for Twitch livestreams/VODs.

Homepage URL: https://github.com/AlyoshaVasilieva/luminous-ttv
Licenses: GPL-3.0-Only, MIT

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lu/luminous-ttv/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "luminous-ttv";
+  version = "0.5.7";
+
+  src = fetchFromGitHub {
+    owner = "AlyoshaVasilieva";
+    repo = pname;
+    rev = "v{version}";
```
Comment (SigmaSquadron) [inline pkgs/by-name/lu/luminous-ttv/package.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/lu/luminous-ttv/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    hash = "sha256-uaq5uiSp4lci27BxtqjdtUeiJvXhWo25lfFE+dQys6Y=";
+  };
+
+  cargoHash = "sha256-jbtHxarRQ8gpCBc/HZWSnkzMrlMMltpknUBV1SGVq/I=";
+
+  meta = {
+    description = "A Rust server to retrieve and relay a playlist for Twitch livestreams/VODs.";
+    homepage = "https://github.com/AlyoshaVasilieva/{pname}";
+    downloadPage = "https://github.com/AlyoshaVasilieva/{pname}/releases/latest";
+    changelog = "https://github.com/AlyoshaVasilieva/{pname}/releases/tag/v{version}";
```
Comment (SigmaSquadron) [inline pkgs/by-name/lu/luminous-ttv/package.nix:20]:
```suggestion
    homepage = "https://github.com/AlyoshaVasilieva/${pname}";
    downloadPage = "https://github.com/AlyoshaVasilieva/${pname}/releases/latest";
    changelog = "https://github.com/AlyoshaVasilieva/${pname}/releases/tag/v${version}";
```

**Review by SigmaSquadron — CHANGES_REQUESTED** [review]
Please format the changed files with `nixfmt-rfc-style` and set your `editorconfig` settings to add a newline at the end of the file. 

(See [CONTRIBUTING.md#syntax](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax))

File: pkgs/by-name/lu/luminous-ttv/package.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/by-name/lu/luminous-ttv/package.nix:20]: Better: do not use "${pname}"

https://github.com/NixOS/nixpkgs/pull/278611

**Review by SigmaSquadron — CHANGES_REQUESTED** [review]
Please squash the "follow code conventions" commit into the main `init` one. There should only be two commits for this PR: your maintainership one, and the package initialisation.

**Comment by alexanderjkslfj** [comment]
I tried to squash the commits into one, but clearly I failed. I'm not well-versed in using git.
Instead of trying again I think it's easier to just cancel this PR and to start a fresh one where I do it correctly from the start.

**Comment by SigmaSquadron** [comment]
No need. The easiest way to do this (without rebasing) is to reset your changes and re-commit.

Run the following:
```bash
git reset --soft HEAD~3 # This will undo the last three commits, including the original init one.
git commit -m "luminous-ttv: init at 0.5.7" -s -S # Commit all changes together.
```

**Review by SigmaSquadron — APPROVED** [review]
Excellent work!

**Review by AndersonTorres — APPROVED** [review]
Result of `nixpkgs-review pr 335481` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>luminous-ttv</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**